lib/Interpreter/CXCppInterOp.cpp

```diff
@@ -0,0 +1,175 @@
+      dir, isUser, prepend);
+}
+
+void clang_interpreter_addIncludePath(CXInterpreter I, const char* dir) {
+  static_cast<Cpp::Interpreter*>(I)->AddIncludePath(dir);
+}
+
+const char* clang_interpreter_getResourceDir(CXInterpreter I) {
+  return static_cast<Cpp::Interpreter*>(I)
+      ->getCI()
+      ->getHeaderSearchOpts()
+      .ResourceDir.c_str();
+}
+
+enum CXErrorCode clang_interpreter_declare(CXInterpreter I, const char* code,
+                                           bool silent) {
+  auto interp = static_cast<Cpp::Interpreter*>(I);
+  auto& diag = interp->getSema().getDiagnostics();
+
+  bool is_silent_old = diag.getSuppressAllDiagnostics();
+
+  diag.setSuppressAllDiagnostics(silent);
+  auto result = interp->declare(code);
+  diag.setSuppressAllDiagnostics(is_silent_old);
+
+  if (result)
+    return CXError_Failure;
+  else
+    return CXError_Success;
+}
+
+enum CXErrorCode clang_interpreter_process(CXInterpreter I, const char* code) {
+  auto result = static_cast<Cpp::Interpreter*>(I)->process(code);
+  if (result)
+    return CXError_Failure;
+  else
+    return CXError_Success;
+}
+
+CXValue clang_createValue() {
+#ifdef USE_CLING
+  auto val = std::make_unique<cling::Value>();
+#else
+  auto val = std::make_unique<clang::Value>();
+#endif // USE_CLING
+
+  return val.release();
+}
+
+void clang_value_dispose(CXValue V) {
+#ifdef USE_CLING
+  delete static_cast<cling::Value*>(V);
+#else
+  delete static_cast<clang::Value*>(V);
+#endif // USE_CLING
+}
+
+enum CXErrorCode clang_interpreter_evaluate(CXInterpreter I, const char* code,
+                                            CXValue V) {
+#ifdef USE_CLING
+  auto val = static_cast<cling::Value*>(V);
+#else
+  auto val = static_cast<clang::Value*>(V);
+#endif // USE_CLING
+
+  auto result = static_cast<Cpp::Interpreter*>(I)->evaluate(code, *val);
+  if (result)
+    return CXError_Failure;
+  else
+    return CXError_Success;
+}
+
+CXString clang_interpreter_lookupLibrary(CXInterpreter I,
+                                         const char* lib_name) {
+  return createCXString(static_cast<Cpp::Interpreter*>(I)
+                            ->getDynamicLibraryManager()
+                            ->lookupLibrary(lib_name));
+}
+
+CXInterpreter_CompilationResult
+clang_interpreter_loadLibrary(CXInterpreter I, const char* lib_stem,
+                              bool lookup) {
+  return static_cast<CXInterpreter_CompilationResult>(
+      static_cast<Cpp::Interpreter*>(I)->loadLibrary(lib_stem, lookup));
+}
+
+void clang_interpreter_unloadLibrary(CXInterpreter I, const char* lib_stem) {
+  static_cast<Cpp::Interpreter*>(I)->getDynamicLibraryManager()->unloadLibrary(
+      lib_stem);
+}
+
+CXString clang_interpreter_searchLibrariesForSymbol(CXInterpreter I,
+                                                    const char* mangled_name,
+                                                    bool search_system) {
+  return createCXString(
```

Comment:
Is it OK to keep a `StringRef` in the dynamic library manager to store the output? For the C++ interface, returning a `std::string` is fine,  but it's not C-friendly.